frontend/test/e2e/caselaw/fixtures.ts

```diff
@@ -30,28 +36,54 @@ type MyFixtures = {
   prefilledDocumentUnitWithManyReferences: DocumentUnit
 }
 
-export const caselawTest = test.extend<MyFixtures>({
-  documentNumber: async ({ request, context }, use) => {
-    const cookies = await context.cookies()
-    const csrfToken = cookies.find((cookie) => cookie.name === "XSRF-TOKEN")
-    const response = await request.put(`/api/v1/caselaw/documentunits/new`, {
+/**
+ * The deletion fails if two doc units are deleted at the same time that have a common relationship (e.g. duplicate relation). As we do not lock the entity, it might try to delete a stale reference and fail. This is very unlikely to happen in production. If so, it would succeed on retry as well.
+ */
+async function deleteWithRetry(
+  request: APIRequestContext,
+  uuid: string,
+  csrfToken: Cookie | undefined,
+  documentNumber: string,
+) {
+  const deleteResponse = await request.delete(
+    `/api/v1/caselaw/documentunits/${uuid}`,
+    {
       headers: { "X-XSRF-TOKEN": csrfToken?.value ?? "" },
-    })
-
-    const { uuid, documentNumber } = await response.json()
+    },
+  )
 
-    await use(documentNumber)
-    const deleteResponse = await request.delete(
+  if (!deleteResponse.ok()) {
+    console.log(
+      `Deletion for ${documentNumber} failed with status ${deleteResponse.status()}, retrying deletion...`,
+    )
+    const retryWaitDuration = Math.floor(Math.random() * 1900) + 100
```

Comment:
Super Umsetzung @kiwikern! Für diese magic numbers fänd ich noch einen Kommentar toll, z.B. `retry at a random time between 0.1 and 2 seconds`